src/auth/authenticator.rs

```diff
@@ -52,3 +66,42 @@ impl AuthenticationError {
         AuthenticationError::ImplPropagated(s.into(), Some(Box::new(source)))
     }
 }
+
+/// Credentials passed to an [Authenticator](crate::auth::Authenticator)
+/// It is guaranteed that it won't happen that both `certificate_chain` and `password`
```

Comment:
Could you elaborate a bit more on how this scenario is impossible to ever occur?

Reply:
We'll the logic to be written in libunftp should ensure that :-). This is just documentation for the implementer to say that they can assume this is the case. We could also create an enum for this if it feels clunky.

Reply:
Currently only password will be passed and it will always be passed.

Reply:
Okay, in that case I think the comment could be clarified a bit more. Because this could be read incorrectly as "we make sure that this never happens". So instead of saying "it is guaranteed" we might need to say "the implementor of the authenticator is responsible for making sure that"... or something

Reply:
Like
`/// The authenticator implementation makes sure that `certificate_chain` and `password` won't both be None`

Reply:
Okay then I misunderstood you, and yes that is how I initially interpreted the message too. But I just didn't read the code that guarantees this. LGTM

Reply:
How about:

```
/// Credentials passed to an [Authenticator](crate::auth::Authenticator)
///
/// [Authenticator](crate::auth::Authenticator) implementations can assume that either `certificate_chain` or `password`
/// will not be `None`.

```

Reply:
yes better....